### PR TITLE
Remove redundant fakecaclient.

### PR DESCRIPTION
**Pull request by myidpt**
Somehow we have FakeProtocol and FakeCAClient doing the same thing, but only FakeProtocol is used. This PR merges them.

### CODE REVIEW

**Comment by incfly** [comment]
I'm a bit of inclined to leave the FakeProtocolClient in the protocol.go? No need to move a mock package. Comparision:

- `mock.NewProtocolClient`
- `protocol.NewFakeClient`

The second one is clearer given the package name itself has more information.

But not a big thing and if elsewhere we always have mock in separate package, we can just leave as it is.

File: security/pkg/caclient/protocol/mock/fakeprotocol.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
+// Copyright 2017 Istio Authors
```
[the remark this answers is not in the document]
Comment (wattli) [reply security/pkg/caclient/protocol/mock/fakeprotocol.go]: +1

**Comment by myidpt** [comment]
@incfly That's a good point. I dig into this a bit and found the standard is to define the mock function where it is used:
https://github.com/golang/go/wiki/CodeReviewComments#interfaces
I'm going to follow the standard.

**Comment by incfly** [comment]
/approve

Thanks for the cleaning up!

**Comment by myidpt** [comment]
@incfly It seems sub-optimal to copy the mock function to all the places using it. There are three tests using it, and I don't like having the code with too much redundancy. I will keep it this way :)

**Pull request merged**